packages/runtime/src/style.tsx

```diff
@@ -23,24 +24,38 @@ export default function Style(props: StyleProps) {
     props.children.forEach(analyzeCssInDev);
   }
 
-  const rules = props.children.filter((sheet) => {
-    if (inserted[sheet]) {
-      return false;
-    }
+  const sheets = props.children;
+
+  if (!sheets.length) {
+    return null;
+  }
+
+  if (isNodeEnvironment()) {
+    const filteredSheets = sheets.filter((sheet) => {
+      if (inserted[sheet]) {
+        return false;
+      }
+
+      inserted[sheet] = true;
 
-    inserted[sheet] = true;
+      return true;
+    });
 
-    return true;
-  });
+    if (filteredSheets.length) {
+      const rulesGroupedByBucket = groupByBucket(filteredSheets);
 
-  if (rules.length) {
-    if (isNodeEnvironment()) {
-      return <style nonce={props.nonce}>{rules}</style>;
-    } else {
-      // Keep re-assigning over ternary because it's smaller
-      stylesheet = stylesheet || createStyleSheet(props);
-      rules.forEach(stylesheet);
+      return buckets
+        .filter((bucket) => !!rulesGroupedByBucket[bucket])
+        .map((bucket) => (
+          <style key={bucket} {...{ [getCompiledAttr(bucket)]: '' }} nonce={props.nonce}>
```

Comment:
using spread here ends up adding boilerplate code to support it - ideally let's come up with an alternate solution